code/iaas/model/src/main/java/io/cattle/platform/core/cleanup/TableCleanup.java

```diff
@@ -169,6 +172,37 @@ private void cleanupServiceEventTable(Date cutoff) {
             log.info("[Rows Deleted] service_event={}", rowsDeleted);
         }
     }
+    
+    private void cleanupMountTable(Date cutoff) {
+    	List<Long> purgeInstanceIds = create()
+    			.select(INSTANCE.ID)
+    			.from(INSTANCE)
+    			.where(INSTANCE.STATE.eq("purged"))
+    			.fetch().into(Long.class);
+    	List<Long> localOrHostBindMountVolumeIds = create()
+    			.select(VOLUME.ID)
+    			.from(VOLUME)
+    			.where(VOLUME.DATA.like("%\"isHostPath\":true%"))
+    			.or(VOLUME.DATA.like("%\"driver\":\"local\"%"))
+    			.fetch().into(Long.class);
+    	ResultQuery<Record1<Long>> ids = create()
+    			.select(MOUNT.ID)
+    			.from(MOUNT)
+    			.where(MOUNT.INSTANCE_ID.in(purgeInstanceIds))
+    			.and(MOUNT.VOLUME_ID.in(localOrHostBindMountVolumeIds))
+    			.and(MOUNT.STATE.eq("inactive"));
+    	List<Long> toDelete = null;
+        int rowsDeleted = 0;
+        while ((toDelete = ids.fetch().into(Long.class)).size() > 0) {
```

Comment:
This while loop doesnt really make sense the way you have it written. It would only ever run once bc on your first run, you always get and delete ALL the ids that the `ids` query will find.